### ISSUE TITLE
Missing mandatory field "SummaryContent" on Situation for TripRequest

### ISSUE THREAD

**Issue by michaelruppen** [openTdataCH/ojp-sdk#128]
Mandatory field SummaryContent is missing on TextualContent [Gist](https://gist.github.com/michaelruppen/e4114acca7973426a711102971cacf8d)

![Bildschirmfoto 2024-09-09 um 16 33 19](https://github.com/user-attachments/assets/0c7990ab-7026-4dda-8761-0771e8b6f306)

As checked with @TO-mdv the input data was already bad. In future situtations with missing mandatory fields get discarded.

**Comment by TO-mdv** [comment]
@ue71603 gemäss xsd Validierung in Altova ist das Element nicht mandatory, hier eine valie Response mit einer Störungsmeldung:
![Image](https://github.com/user-attachments/assets/0d1edf07-c6d5-41f4-8a77-5285ab9e548e)

Gemäss Dokue von @michaelruppen ist es jedoch mandatory: https://vdvde.github.io/OJP/develop/documentation-tables/siri.html#type_siri__TextualContentStructure

Kannst du mir sagen, was wir hier falsch interpretieren?

**Comment by ue71603** [comment]
PtSituationElement/PtSituationBodyGroup/PublishingActions/PublishingAction/PassengerInformationAction/TextualContent enthält ein obligatorisches Element SummaryContent

Was ich aber bei Dir sehe, ist, dass Dein Ausschnitt - @TO-mdv - kommt gar nie bei PublishingAction vorbei (was es u.U. nicht sollte)
Du endest in PtSituationElement/PtSituationBodyGroup/PublishingActions/PublishingAction/Affects und bist fertig.

Da PublishingAction optional ist, gibt XMLSpy sein ok.

Btw: Leere Elemente wie <siri:Affects/> machen keine Freude in der Auswertung.

Die Frage ist, was Ihr wirklich hättet ausfüllen wollen.

**Comment by TO-mdv** [comment]
Analyse, welche Elemente nicht XSD konform sind:

- wir liefern beide Elemente <siri:AlertCause> und <siri:UnknownReason>, nur eines davon ist erlaubt
- unter Element <siri:VehicleJourneys/> fehlt <siri:AffectedVehicleJourney>, bzw. das Element VehicleJourneys könnte auch ganz weggelassen werden
- Element <siri:PublishAtScope> fehlt, inkl. <siri:ScopeType> und <siri:Affects>
- Element <siri:ActionRef> fehlt
- Element <siri:Perspective> fehlt
- unter Element <siri:SummaryContent> fehlt der <siri:SummaryText>
- unter Element <siri:DescriptionContent> fehlt der <siri:DescriptionText>
- unter Element <siri:ConsequenceContent> fehlt <siri:ConsequenceText>
- unter Element <siri:RecommendationContent> fehlt <siri:RecommendationText>


![Image](https://github.com/user-attachments/assets/029c74df-967c-42c7-8cef-95abb14dd7d7)

**Comment by ue71603** [comment]
@TO-mdv müssen wir das jetzt mit Roger anschauen oder könnt Ihr das fixen?

**Comment by TO-mdv** [comment]
@ue71603 Nein, ich bin mit @herlitze soeben nochmals die Struktur durchgegangen, wir passen sie für die OJP2.0 Ausgabe an, um wieder XSD konform zu sein.

Wenn halt Elemente in SIRI-SX fehlen, werden wir erstmals einfach leere Felder ausgeben.

**Comment by TO-mdv** [comment]
Folgende Anpassungen haben wir im Rahmen von [SH-3680](https://youtrack.mentz.net/issue/SH-3680) OJP2.0 Störungsmeldungen / PtSituation XSD konform vorgenommen:

- /OJP/OJPResponse/siri:ServiceDelivery/OJPTripDelivery/TripResponseContext/Situations/PtSituation/siri:UnknownReason -> entfernt, da nicht zulässig

- /OJP/OJPResponse/siri:ServiceDelivery/OJPTripDelivery/TripResponseContext/Situations/PtSituation/siri:Affects -> je nachdem, ob VehicleJourney oder StopPoint betroffen, wird nur noch die jeweilige mandatory Struktur ausgegeben

- /OJP/OJPResponse/siri:ServiceDelivery/OJPTripDelivery/TripResponseContext/Situations/PtSituation/siri:PublishingActions/siri:PublishingAction/siri:PublishAtScope/siri:ScopeType -> wird nun immer mit "unknown" ausgegeben

- /OJP/OJPResponse/siri:ServiceDelivery/OJPTripDelivery/TripResponseContext/Situations/PtSituation/siri:PublishingActions/siri:PublishingAction/siri:PassengerInformationAction/siri:ActionRef -> wird nun immer leer ausgegeben

- /OJP/OJPResponse/siri:ServiceDelivery/OJPTripDelivery/TripResponseContext/Situations/PtSituation/siri:PublishingActions/siri:PublishingAction/siri:PassengerInformationAction/siri:RecordedAtTime -> wenn in der Meldung nicht vorhanden, wird Startdate 1900-... gesetzt

- /OJP/OJPResponse/siri:ServiceDelivery/OJPTripDelivery/TripResponseContext/Situations/PtSituation/siri:PublishingActions/siri:PublishingAction/siri:PassengerInformationAction/siri:Perspective -> wird korrekt gesetzt gemäss Enums

- /OJP/OJPResponse/siri:ServiceDelivery/OJPTripDelivery/TripResponseContext/Situations/PtSituation/siri:PublishingActions/siri:PublishingAction/siri:PassengerInformationAction/siri:TextualContent/siri:SummaryContent/siri:SummaryText -> wenn kein Text vorhanden, wird n/a ausgegeben

- restliche TextualContent Felder werden nur ausgegben, wenn Text vorhanden

**Comment by r3to** [comment]
@TO-mdv Kann das geschlossen werden?

**Comment by TO-mdv** [comment]
@r3to richtig, danke für den Hinweis.
